apis/apiextensions/v1/composition_patches_test.go

```diff
@@ -0,0 +1,165 @@
+			reason: "FromCompositeFieldPath patch with FromFieldPath set should be valid",
+			args: args{
+				patch: &Patch{
+					Type:          PatchTypeFromCompositeFieldPath,
+					FromFieldPath: pointer.String("spec.forProvider.foo"),
+				},
+			},
+		},
+		"FromCompositeFieldPathWithInvalidTransforms": {
+			reason: "FromCompositeFieldPath with invalid transforms should return error",
+			args: args{
+				patch: &Patch{
+					Type:          PatchTypeFromCompositeFieldPath,
+					FromFieldPath: pointer.String("spec.forProvider.foo"),
+					Transforms: []Transform{
+						{
+							Type: TransformTypeMath,
+							Math: nil,
+						},
+					},
+				},
+			},
+			want: want{
+				err: &field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "transforms[0].math",
+				},
+			},
+		},
+		"InvalidFromCompositeFieldPathMissingFromFieldPath": {
+			reason: "Invalid FromCompositeFieldPath missing FromFieldPath should return error",
+			args: args{
+				patch: &Patch{
+					Type:          PatchTypeFromCompositeFieldPath,
+					FromFieldPath: nil,
+				},
+			},
+			want: want{
+				err: &field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "fromFieldPath",
+				},
+			},
+		},
+		"InvalidFromCompositeFieldPathMissingToFieldPath": {
+			reason: "Invalid ToCompositeFieldPath missing ToFieldPath should return error",
+			args: args{
+				patch: &Patch{
+					Type:        PatchTypeToCompositeFieldPath,
+					ToFieldPath: nil,
+				},
+			},
+			want: want{
+				&field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "fromFieldPath",
+				},
+			},
+		},
+		"InvalidPatchSetMissingPatchSetName": {
+			reason: "Invalid PatchSet missing PatchSetName should return error",
+			args: args{
+				patch: &Patch{
+					Type: PatchTypePatchSet,
+				},
+			},
+			want: want{
+				err: &field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "patchSetName",
+				},
+			},
+		},
+		"InvalidCombineMissingCombine": {
+			reason: "Invalid Combine missing Combine should return error",
+			args: args{
+				patch: &Patch{
+					Type: PatchTypeCombineToComposite,
+				},
+			},
+			want: want{
+				err: &field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "combine",
+				},
+			},
+		},
+		"InvalidCombineMissingToFieldPath": {
+			reason: "Invalid Combine missing ToFieldPath should return error",
+			args: args{
+				patch: &Patch{
+					Type: PatchTypeCombineToComposite,
+					Combine: &Combine{
+						Variables: []CombineVariable{
+							{
+								FromFieldPath: "spec.forProvider.foo",
+							},
+						},
+					},
+					ToFieldPath: nil,
+				},
+			},
+			want: want{
+				err: &field.Error{
+					Type:  field.ErrorTypeRequired,
+					Field: "toFieldPath",
+				},
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			err := tc.args.patch.Validate()
+			if diff := cmp.Diff(
+				tc.want.err,
+				err,
+				cmpopts.IgnoreFields(field.Error{}, "Detail", "BadValue"),
```

Comment:
nit: wrapping code lines sometimes makes the code less readable, so prefer not to do it unless it is really too long.

Reply:
Just out of curiosity, any heuristic to define too long? I used to work on projects in which the 120 characters limit was enforced by the linter itself, so my "too long" might be a little bit on the strict side.

Reply:
Nonetheless, put all on a single line as suggested.

Reply:
For documentation (.md files) and comments, we wrap at 80 character.
But we don't have a limit/rule for the code itself and rather optimize for readability.